lib/api.ts

```diff
@@ -4,58 +4,55 @@ import invariant from 'tiny-invariant';
 import Poll, { PartialPoll } from '../types/poll';
 import { CMSProposal } from '../types/proposal';
 import BlogPost from '../types/blogPost';
 import { parsePollMetadata } from './polling/parser';
+import { Octokit } from '@octokit/core';
 
 export async function getExecutiveProposals(): Promise<CMSProposal[]> {
-  if (process.env.USE_FS_CACHE) {
-    const cachedProposals = fsCacheGet('proposals');
-    if (cachedProposals) return JSON.parse(cachedProposals);
-  } else if (process.env.NEXT_PUBLIC_USE_MOCK || isTestnet()) return require('../mocks/proposals.json');
   const network = getNetwork();
-  invariant(network in CMS_ENDPOINTS, `no cms endpoint known for network ${network}`);
-  const topics = await (await fetch(CMS_ENDPOINTS[network].allTopics)).json();
-  const spells = await (await fetch(CMS_ENDPOINTS[network].allSpells)).json();
-  let proposals: Array<any> = topics
-    .filter(topic => topic.active)
-    .filter(topic => !topic.govVote)
-    .map(topic => topic.proposals)
-    .flat();
-
-  spells.forEach(spell => {
-    spell.address = spell.source;
-    delete spell.source;
-  });
 
-  proposals.forEach(proposal => {
-    proposal.proposalBlurb = proposal.proposal_blurb;
-    proposal.address = proposal.source;
-    proposal.active = true;
-    delete proposal.proposal_blurb;
-    delete proposal.source;
-  });
+  if (isTestnet()) return [];
+
+  const proposalIndex = await (await fetch(EXEC_PROPOSAL_INDEX)).json();
 
-  const oldSpells = spells
-    .filter(
-      spell => proposals.findIndex(proposal => proposal.address === spell.address) === -1 // filter out active spells
-    )
-    .sort((a, b) => new Date(b.date).getTime() - new Date(a.date).getTime());
-
-  oldSpells.forEach(spell => {
-    spell.active = false;
-    spell.proposalBlurb = spell.proposal_blurb;
-    spell.key = spell._id;
-    delete spell.proposal_blurb;
-    delete spell._id;
+  const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
+
+  const { data } = await octokit.request('GET /repos/{owner}/{repo}/contents/{path}', {
+    owner: 'tyler17', //FIXME change to makerdao before merging
+    repo: 'community',
+    path: 'governance/votes'
   });
+  const githubResponse = Array.isArray(data) ? data : [];
+  const proposalUrls = githubResponse.filter(x => x.type === 'file').map(x => x.download_url);
+
+  let proposals: CMSProposal[] = [];
+  for (const proposalLink of proposalUrls) {
+    if (!proposalLink) continue;
+    const proposalDoc = await (await fetch(proposalLink)).text();
+    const {
+      content,
+      data: { title, summary, address, date }
+    } = matter(proposalDoc);
+
+    invariant(content && title && summary && address && date, 'Invalid proposal document');
+    proposals.push({
+      about: content,
+      title,
+      proposalBlurb: summary,
+      key: slugify(title),
+      address: address,
+      date: String(date),
+      active: proposalIndex[network] ? proposalIndex[network].includes(proposalLink) : true
```

Comment:
so would the `else` case here mainly trigger when we're on a testchain?

Reply:
yeah exactly